php_phongo.c

```diff
@@ -1281,148 +1295,186 @@ static bool php_phongo_apply_wc_options_to_uri(mongoc_uri_t *uri, bson_t *option
 	return true;
 } /* }}} */
 
-static bool php_phongo_apply_ssl_opts(mongoc_client_t *client, zval *zdriverOptions TSRMLS_DC)
+static inline char *php_phongo_fetch_ssl_opt_string(zval *zoptions, const char *key, int key_len)
 {
-	mongoc_ssl_opt_t ssl_opt = {0};
-	zval *zoptions;
-	zend_bool free_pem_file = 0, free_pem_pwd = 0, free_ca_file = 0, free_ca_dir = 0, free_crl_file = 0;
-	int plen;
-	bool valid_options;
+	int        plen;
+	zend_bool  pfree;
+	char      *pval, *value;
 
-	if (!zdriverOptions) {
-		return false;
-	}
-
-	zoptions = zdriverOptions;
-
-	/* Use SSL context options if provided for backwards compatibility */
-	if (php_array_existsc(zdriverOptions, "context")) {
-		php_stream_context *context;
-		zval *zcontext;
+	pval = php_array_fetchl_string(zoptions, key, key_len, &plen, &pfree);
+	value = pfree ? pval : estrndup(pval, plen);
 
-		zcontext = php_array_fetchc(zdriverOptions, "context");
+	return value;
+}
 
-		context = php_stream_context_from_zval(zcontext, 1);
+static mongoc_ssl_opt_t *php_phongo_make_ssl_opt(zval *zoptions TSRMLS_DC)
+{
+	mongoc_ssl_opt_t *ssl_opt;
 
-		if (!context) {
-			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "\"context\" driver option is not a valid Stream-Context resource");
-			return false;
-		}
+	if (!zoptions) {
+		return NULL;
+	}
 
-#if PHP_VERSION_ID >= 70000
-		zoptions = php_array_fetchc_array(&context->options, "ssl");
-#else
-		zoptions = php_array_fetchc_array(context->options, "ssl");
+#if defined(MONGOC_ENABLE_SSL_SECURE_CHANNEL) || defined(MONGOC_ENABLE_SSL_SECURE_TRANSPORT)
+	if (php_array_existsc(zoptions, "ca_dir") || php_array_existsc(zoptions, "capath")) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "\"ca_dir\" and \"capath\" options are not supported by Secure Channel and Secure Transport");
```

Comment:
> Perhaps we should have another log handler to use when mongodb.debug is disabled so we can still provide exceptions for ERROR and CRITICAL messages while ignoring lesser logs.

@derickr made a good point that this would simply cause existing code to start throwing new or different exceptions out of the blue. Grepping the libmongoc code, SSL option parsing appears to be one of the few places where an ERROR is logged (to report an option being ignored) and the code continues doing its thing without immediately returning. I suppose we'll stick with checking for these during PHPC option parsing.